Replace `node-sass` with `dart-sass`?

## Is your feature request related to a problem? Please describe.
I seem to have a lot of trouble using `node-sass` on my MacBook - perhaps related to Apple Silicon (M1 processor)?

## Describe the solution you'd like
I tested by replacing `node-sass` in `devDependencies` with `sass@"1.49.9"` (https://github.com/sass/dart-sass) and it seemed to work OK.

When searching online, I saw others mentioning that they gave up trying to make `node-sass` work on ARM architecture, and just replaced their dependency with `dart-sass` instead.

`node-sass` even has this warning in their readme:

> Warning: [LibSass and Node Sass are deprecated](https://sass-lang.com/blog/libsass-is-deprecated). While they will continue to receive maintenance releases indefinitely, there are no plans to add additional features or compatibility with any new CSS or Sass features. Projects that still use it should move onto [Dart Sass](https://sass-lang.com/dart-sass).

https://github.com/sass/node-sass#node-sass

## Describe alternatives you've considered
I am not sure what to do, the error message is cryptic and I'm not able to understand what is the problem.

## Additional context
Here is a part of the error I see when trying to build the assets on my machine. I believe this is when it's trying to get the `node-sass` dependency, so even before it reaches any webpack step at all (the command I ran was `npm install`):

```
npm ERR! /Users/thomasrambo/.node-gyp/17.8.0/include/node/v8-internal.h:563:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
npm ERR!             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
npm ERR!                                 ~~~~~^~~~~~~~~~~
npm ERR!                                      remove_cv
npm ERR! /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:710:50: note: 'remove_cv' declared here
npm ERR! template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
npm ERR!                                                  ^
npm ERR! 1 error generated.
npm ERR! make: *** [Release/obj.target/binding/src/binding.o] Error 1
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/Users/thomasrambo/Documents/devel/invenio3/invenio-assets/invenio_assets/assets/node_modules/node-gyp/lib/build.js:262:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Darwin 21.2.0
npm ERR! gyp ERR! command "/opt/homebrew/Cellar/node/17.8.0/bin/node" "/Users/thomasrambo/Documents/devel/invenio3/invenio-assets/invenio_assets/assets/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! gyp ERR! cwd /Users/thomasrambo/Documents/devel/invenio3/invenio-assets/invenio_assets/assets/node_modules/node-sass
npm ERR! gyp ERR! node -v v17.8.0
npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! gyp ERR! not ok
npm ERR! Build failed with error code: 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/thomasrambo/.npm/_logs/2022-03-28T17_28_19_551Z-debug-0.log
```

--- a/invenio_assets/assets/package.json
+++ b/invenio_assets/assets/package.json
@@ -47,7 +47,7 @@
     "less": "^3.12.2",
     "less-loader": "6.2.0",
     "mini-css-extract-plugin": "0.10.0",
-    "node-sass": "^4.12.0",
+    "sass": "^1.50.0",
     "optimize-css-assets-webpack-plugin": "5.0.3",
     "ora": "^5.0.0",
     "patch-package": "^6.2.2",

